### ISSUE TITLE
I lost all of my data, including the profile

### ISSUE THREAD

**Issue by Remo253** [medicalid/issues#235]
I just realized that the Medical ID icon wasn't appearing on my screen. I opened the app and it acts as if it's first time setup. None of the data I'd entered months ago is there. It worked fine for a very long time. The only issue I had was trying to pin the icon to one place on the screen, it kept moving around.

I use Google Fi, riding the T-Mobile network. There are regular updates and it may have disappeared with one of those. Unfortunately I can say when it disappeared. I was looking for something else today in the list of apps, saw it's listing and realized the icon wasn't showing any more.

Samsung Note 9, Android 10, Kernal version 4.9.186-18803994, Medical ID 7.10.0

**Comment by laurentpellegrino** [comment]
Thanks for reaching out.

App data is erased when you uninstall the app or clear app data (e.g. with a cleanup utility app).

The premium version of the app includes a backup feature. However, to give you full control over your data, it must be triggered manually. You can read more here:

https://medicalid.app/faq#backups_location

This should be discussed directly in the app so that people are not surprised. I will consider an improvement regarding this area.

**Comment by Remo253** [comment]
I didn't uninstall it. It was still there, just had no data, as if it had not been run after installation. While I've cleared data and cache for specific apps I have never used a third party app to clear storage. I don't store much, right now I'm at 70% free. I have used the Android default Storage manager found in Settings but that's only to ID Photos and Videos that could be deleted.

When I get an OS update it goes through a process, I don't recall the exact verbiage, where it's checking all the apps, "something something app 30 of 450, 31 of 450, etc". Could that process remove the data?

Btw, I really like the App, told some friends about it, and just upgraded to the paid version. I've re-entered the data, did the backup and moved it off the phone. An auto backup every x number of days would be a nice feature, especially if you could specify a cloud service.

**Comment by laurentpellegrino** [comment]
> When I get an OS update it goes through a process, I don't recall the exact verbiage, where it's checking all the apps, "something something app 30 of 450, 31 of 450, etc". Could that process remove the data?

No, this does not remove app data.

> While I've cleared data and cache for specific apps I have never used a third party app to clear storage

If you have cleared data and cache for Medical ID, this erased your medical profiles.

> An auto backup every x number of days would be a nice feature, especially if you could specify a cloud service.

I agree with you, that's definitively a great solution. Hope to find the time to implement it in the future.

**Comment by Remo253** [comment]
"While I've cleared data and cache for specific apps....................."
"If you have cleared data and cache for Medical ID, this erased your medical profiles."

I haven't, I've used that for other apps but never for Medical ID. 

- I didn't uninstall/reinstall Medical ID
- I didn't do anything to clear it's storage with a third party app, none installed, don't use them
- I've never cleared data or cache for Medical ID using Android default tools, I know that would delete my info
- Once setup I hadn't touched it in anyway, besides moving the icon around.
- Nonetheless, data and profile disappeared at some point in the last few months
- The only thing done to the phone in those months was normal updates from the carrier, T-Mobile, and adding/removing a handful of OTHER apps

I take it no one else has reported anything similar? If so I understand you can't ID the issue so take this as a report of an anomalous occurrence.

**Comment by laurentpellegrino** [comment]
@Remo253 Thanks for the additional information.

A few people have reported data loss but after discussion, they admit the Medical ID app was uninstalled or they cleared data indirectly with a cleanup tool. 

As you mentioned, the only remaining possible cause I see is that a device update causes the issue. I have already investigated this months ago but I was not able to reproduce such a problem. I will try again but this could be really device-specific. If I succeed to reproduce or if I receive new reports, I will reopen this issue.
